tests/test_codex_feed_hooks.py

```
@@ -2362,6 +2426,788 @@ def test_non_codex_post_tool_use_keeps_request_input(cli_path: str, root: Path)
+
+
+def test_pi_feed_rejects_unconfirmed_server_ack(cli_path: str, root: Path) -> None:
+    socket_path = root / "cmux-pi-feed-unconfirmed-ack.sock"
+    env = os.environ.copy()
+    for key in ("CMUX_SOCKET", "CMUX_SOCKET_CAPABILITY", "CMUX_SOCKET_PATH", "CMUX_SOCKET_PASSWORD"):
+        env.pop(key, None)
+    env["CMUX_SURFACE_ID"] = FAKE_SURFACE_ID
+    env["CMUX_WORKSPACE_ID"] = FAKE_WORKSPACE_ID
+    payload = {
+        "session_id": "pi-unconfirmed-session",
+        "cwd": "/tmp/pi-unconfirmed-project",
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/tmp/pi-unconfirmed-project"

(S108)